docs/examples/featuresets/reading_datasets.py

```diff
@@ -40,9 +40,47 @@ def func(cls, ts: pd.Series, uids: pd.Series):
 # /docsnip
 
 
+# docsnip derived_extractors
+@meta(owner="data-science-team@fennel.ai")
+@featureset
+class Request:
+    user_id: int = feature(id=1)
+
+
+@meta(owner="data-science-team@fennel.ai")
+@featureset
+class UserFeaturesDerived:
+    uid: int = feature(id=1).extract(feature=Request.user_id)
+    name: str = feature(id=2).extract(field=User.name, default="Unknown")
+
+
+# /docsnip
+
+
+# docsnip derived_extractor_with_provider
+@meta(owner="data-science-team@fennel.ai")
+@featureset
+class Request2:
+    uid: int = feature(id=1)
+
+
+@meta(owner="data-science-team@fennel.ai")
+@featureset
+class UserFeaturesDerived2:
+    name: str = feature(id=1).extract(
+        field=User.name, provider=Request2, default="Unknown"
```

Comment:
@aditya-nambiar here's a client side example with the provider feature, there are also error cases with the provider feature tested in `test_derived_extractors`